activerecord/test/cases/finder_test.rb

```diff
@@ -1375,7 +1375,14 @@ def test_with_limiting_with_custom_select
       limit: 3, order: "posts.id"
     ).to_a
     assert_equal 3, posts.size
-    assert_equal [0, 1, 1], posts.map(&:author_id).sort
+    assert_equal [1, 1, nil], posts.map(&:author_id)
```

Comment:
I'm kind of confused here, this means a change of behavior/results?

Reply:
Actually yes.

Before this fix, eager load works differently with natural load and preload for now.
Even if `select: 'posts.*, authors.id as "author_id"'` is specified, eager load always get original column values (i.e. mostly ignore the select).

https://github.com/rails/rails/blob/b846c08dcf0396f18eaa85080ee383ad3951429f/activerecord/test/fixtures/posts.yml#L19-L24

https://github.com/rails/rails/blob/b846c08dcf0396f18eaa85080ee383ad3951429f/activerecord/test/cases/finder_test.rb#L1372-L1379

The purpose of this PR is to make the eager load not ignore select, so I've tried to justify it with the following test.

https://github.com/rails/rails/blob/b846c08dcf0396f18eaa85080ee383ad3951429f/activerecord/test/cases/finder_test.rb#L1381-L1386